File: calculaPreco.java
public class Produto {

    String nome;
    double preco;
    double desconto;

    public Produto(String nome, double preco, double desconto) {
        this.nome = nome;
        this.preco = preco;
        this.desconto = desconto;
    }

    public double calcularPrecoFinal() {
        return preco - (preco * desconto / 100);
    }

    public static void main(String[] args) {
        Produto produto = new Produto("Fone de ouvido", 200, 25);

        // Exibe o preço inicial
        System.out.printf("Preço inicial de %s: R$ %.2f%n", produto.nome, produto.preco);

        // Exibe o preço final
        System.out.printf("Preço final de %s: R$ %.2f%n", produto.nome, produto.calcularPrecoFinal());
    }
}

File: calculaPreco.js
function calcularPrecoFinal(produto) {
  const desconto = produto.desconto ?? 0;

  // Exibe o preço inicial
  console.log(`Preço inicial de ${produto.nome}: R$ ${produto.preco.toFixed(2)}`);

  const precoFinal = produto.preco - (produto.preco * desconto / 100);

  // Exibe o preço final
  console.log(`Preço final de ${produto.nome}: R$ ${precoFinal.toFixed(2)}`);
}

// Exemplo de uso
const produto1 = {
  nome: "Fone de ouvido",
  preco: 200,
  desconto: 25
};

calcularPrecoFinal(produto1);
